src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1670,11 +1816,46 @@ bool isRefEscape
             // check val escape of receiver if ref-like
             if (receiverOpt?.Type?.IsRefLikeType == true)
             {
+                // Should we also report ErrorCode.ERR_CallArgMixing if CheckValEscape() fails?
                 return CheckValEscape(receiverOpt.Syntax, receiverOpt, scopeOfTheContainingExpression, escapeTo, false, diagnostics);
             }
 
             return true;
 
+            static ParameterSymbol getParameter(ImmutableArray<ParameterSymbol> parameters, ImmutableArray<int> argsToParamsOpt, int argIndex)
+            {
+                var paramIndex = argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex];
+                return parameters[paramIndex];
+            }
+
+            bool considerParameter(ParameterSymbol? parameter)
+            {
+                if (UseUpdatedEscapeRules)
+                {
+                    if (parameter is null)
+                    {
+                        return true;
+                    }
+                    if (parameter.RefKind == RefKind.Out)
```

Comment:
again would like to see us rely on 'Scope' here rather than directly checking for 'RefKind.Out'

Reply:
It looks like I'd added this unnecessarily. Removed.